assets/prometheus-user-workload/service-monitor-thanos-sidecar.yaml

```diff
@@ -20,7 +20,7 @@ spec:
       certFile: /etc/prometheus/secrets/metrics-client-certs/tls.crt
       insecureSkipVerify: false
       keyFile: /etc/prometheus/secrets/metrics-client-certs/tls.key
-      serverName: prometheus-user-workload-thanos-sidecar
+      serverName: prometheus-k8s-thanos-sidecar.openshift-user-workload-monitoring.svc
```

Comment:
thanos-sidecar targets for in-cluster monitoring don't have problem, targets are UP, but thanos-sidecar targets for UWM monitoring are still down, example:
```
        "labels": {
          "container": "kube-rbac-proxy-thanos",
          "endpoint": "thanos-proxy",
          "instance": "10.128.2.22:10902",
          "job": "prometheus-user-workload-thanos-sidecar",
          "namespace": "openshift-user-workload-monitoring",
          "pod": "prometheus-user-workload-1",
          "service": "prometheus-user-workload-thanos-sidecar"
        },
        "scrapePool": "serviceMonitor/openshift-user-workload-monitoring/thanos-sidecar/0",
        "scrapeUrl": "https://10.128.2.22:10902/metrics",
        "globalUrl": "https://10.128.2.22:10902/metrics",
        "lastError": "Get \"https://10.128.2.22:10902/metrics\": x509: certificate is valid for *.prometheus-user-workload-thanos-sidecar.openshift-user-workload-monitoring.svc, *.prometheus-user-workload-thanos-sidecar.openshift-user-workload-monitoring.svc.cluster.local, prometheus-user-workload-thanos-sidecar.openshift-user-workload-monitoring.svc, prometheus-user-workload-thanos-sidecar.openshift-user-workload-monitoring.svc.cluster.local, not prometheus-k8s-thanos-sidecar.openshift-user-workload-monitoring.svc",
        "lastScrape": "2023-02-02T02:31:06.264925913Z",
        "lastScrapeDuration": 0.01067594,
        "health": "down",
        "scrapeInterval": "30s",
        "scrapeTimeout": "10s"
      },
```
tested with PR
```
$ oc -n openshift-user-workload-monitoring get servicemonitor thanos-sidecar -oyaml | grep serverName
      serverName: prometheus-k8s-thanos-sidecar.openshift-user-workload-monitoring.svc
```
in a healthy 4.13 cluster, it is
```
# oc -n openshift-user-workload-monitoring get servicemonitor thanos-sidecar -oyaml | grep serverName
      serverName: prometheus-user-workload-thanos-sidecar.openshift-user-workload-monitoring.svc
```

Reply:
Thanks, @juzhao for the find! Let's see if the third time is the charm ;)